ovos_plugin_manager/templates/phal.py

```diff
@@ -0,0 +1,408 @@
+        self.bus.remove("enclosure.mouth.smile", self.on_smile)
+        self.bus.remove("enclosure.mouth.viseme", self.on_viseme)
+        self.bus.remove("enclosure.mouth.text", self.on_text)
+        self.bus.remove("enclosure.mouth.display", self.on_display)
+        self.bus.remove("enclosure.mouth.events.activate", self._activate_mouth_events)
+        self.bus.remove("enclosure.mouth.events.deactivate", self._deactivate_mouth_events)
+
+        self.bus.remove("enclosure.weather.display", self.on_weather_display)
+
+        self.bus.remove("mycroft.awoken", self.on_awake)
+        self.bus.remove("recognizer_loop:sleep", self.on_sleep)
+        self.bus.remove("speak", self.on_speak)
+        self.bus.remove('recognizer_loop:record_begin', self.on_record_begin)
+        self.bus.remove('recognizer_loop:record_end', self.on_record_end)
+        self.bus.remove('recognizer_loop:audio_output_start', self.on_audio_output_start)
+        self.bus.remove("enclosure.notify.no_internet", self.on_no_internet)
+
+        self._deactivate_mouth_events()
+        self._running = False
+
+    def run(self):
+        ''' start enclosure '''
+        self._running = True
+        while self._running:
+            time.sleep(1)
+
+    # Audio Events
+    def on_record_begin(self, message=None):
+        ''' listening started '''
+        pass
+
+    def on_record_end(self, message=None):
+        ''' listening ended '''
+        pass
+
+    def on_audio_output_start(self, message=None):
+        ''' speaking started '''
+        pass
+
+    def on_audio_output_end(self, message=None):
+        ''' speaking started '''
+        pass
+
+    def on_awake(self, message=None):
+        ''' on wakeup animation '''
+        pass
+
+    def on_sleep(self, message=None):
+        ''' on naptime animation '''
+        # TODO naptime skill animation should be ond here
+        pass
+
+    def on_speak(self, message=None):
+        ''' on speak messages, intended for enclosures that disregard
+        visemes '''
+        pass
+
+    def on_reset(self, message=None):
+        """The enclosure should restore itself to a started state.
+        Typically this would be represented by the eyes being 'open'
+        and the mouth reset to its default (smile or blank).
+        """
+        pass
+
+    # System Events
+    def on_no_internet(self, message=None):
+        """
+
+        Args:
+            message:
+        """
+        pass
+
+    def on_system_reset(self, message=None):
+        """The enclosure hardware should reset any CPUs, etc."""
+        pass
+
+    def on_system_mute(self, message=None):
+        """Mute (turn off) the system speaker."""
+        pass
+
+    def on_system_unmute(self, message=None):
+        """Unmute (turn on) the system speaker."""
+        pass
+
+    def on_system_blink(self, message=None):
+        """The 'eyes' should blink the given number of times.
+        Args:
+            times (int): number of times to blink
```

Comment:
args don't match signature

Reply:
in all these args they are documenting what comes inside the message.data , they are not final and were copied over from original methods (usually from mark1 code)

as you can probably tell i copied over almost every defined event handler for enclosure in mycroft-core

Reply:
Yeah; figured it was copy/paste.. Maybe we should define some standard way to document expected data/context in these docstrings?